app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdEffectHandler.kt

```diff
@@ -28,8 +30,21 @@ class ScanSimpleIdEffectHandler @AssistedInject constructor(
       .addTransformer(ValidateShortCode::class.java, validateShortCode())
       .addConsumer(SendScannedIdentifierResult::class.java, { uiActions.sendScannedId(it.scannedId) }, schedulersProvider.ui())
       .addTransformer(SearchPatientByIdentifier::class.java, searchPatientByIdentifier())
+      .addTransformer(ParseJsonIntoPatientPrefillInfoObject::class.java, parseJsonIntoObject())
       .build()
 
+  private fun parseJsonIntoObject(): ObservableTransformer<ParseJsonIntoPatientPrefillInfoObject, ScanSimpleIdEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .map {
+            val adapter = moshi.adapter(PatientPrefillInfo::class.java)
+            adapter.fromJson(it.text)
+          }
+          .map(::EnteredCodeJsonParsed)
```

Comment:
Json is from the scanned QR code and not entered code, can we rename this data class to reflect that, please?